# FilterMutectCalls ([doc](https://software.broadinstitute.org/gatk/documentation/tooldocs/current/org_broadinstitute_hellbender_tools_walkers_mutect_FilterMutectCalls.php))

filter somatic SNVs and indels called by Mutect2 with contamination table generated by [GetPileupSummaries](../../coverage/get-pileup-summaries) and [CalculateContamination](../../qc/calculate-contamination)

## IO

- *{sample}*.vcf.gz, *{sample}*.contamination.table -> *{sample}*.oncefiltered.vcf.gz

## Example snakemake rule
```python
rule filter_mutect_calls:
    input:
        # Required input.
        vcf = '{sample}.vcf.gz',
        contamination_table = '{sample}.contamination.table',
    output:
        '{sample}.oncefiltered.vcf.gz'
    params:
        # Optional parameters. Omit if unused.
        extra = ''
    threads: 1
    log: 'logs/gatk/filter-mutect-calls/{sample}.log'
    wrapper:
        'http://dohlee-bio.info:9193/gatk/variant-filtering/filter-mutect-calls'
```
